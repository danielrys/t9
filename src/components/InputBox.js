// @flow
import * as React from "react"
import { View, Text, StyleSheet } from "react-native"

// theme
import { Colors, Metrics } from "../themes"

const styles = StyleSheet.create({
  container: {
    borderWidth: Metrics.borders.widths.base,
    borderRadius: Metrics.radii.base,
    borderColor: Colors.border,
    height: Metrics.inputBox.height,
    justifyContent: "center",
  },
  text: {
    fontSize: Metrics.texts.sizes.base,
    color: Colors.textPrimary,
  },
})

type InputBoxProps = {
  numbers: Array<number>,
}

export default class InputBox extends React.PureComponent<InputBoxProps> {
  render() {
    const { numbers = [] } = this.props
    return (
      <View style={styles.container}>
        <Text style={styles.text}>{numbers.join("")}</Text>
      </View>
    )
  }
}
