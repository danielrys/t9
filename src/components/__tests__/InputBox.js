import React from "react"
import renderer from "react-test-renderer"

// components
import InputBox from "../InputBox"

describe("InputBox component", () => {
  it("renders without numbers prop", () => {
    const tree = renderer.create(<InputBox />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders with given numbers prop", () => {
    const tree = renderer
      .create(<InputBox numbers={[1, 2, 8, 6, 4]} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
