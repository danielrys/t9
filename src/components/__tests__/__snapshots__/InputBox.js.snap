// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox component renders with given numbers prop 1`] = `
<View
  style={
    Object {
      "borderColor": "rgb(150,150,150)",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 35,
      "justifyContent": "center",
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(0,0,0)",
        "fontSize": 16,
      }
    }
  >
    12864
  </Text>
</View>
`;

exports[`InputBox component renders without numbers prop 1`] = `
<View
  style={
    Object {
      "borderColor": "rgb(150,150,150)",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 35,
      "justifyContent": "center",
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(0,0,0)",
        "fontSize": 16,
      }
    }
  >
    
  </Text>
</View>
`;
